services/pull/pull.go

```diff
@@ -80,6 +82,42 @@ func NewPullRequest(ctx context.Context, repo *repo_model.Repository, issue *iss
 			assigneeCommentMap[assigneeID] = comment
 		}
 
+		for _, reviewerID := range reviewerIDs {
+			// negative reviewIDs represent team requests
+			if reviewerID < 0 {
+				team, err := organization.GetTeamByID(ctx, -reviewerID)
+				if err != nil {
+					return err
+				}
+				err = issue_service.IsValidTeamReviewRequest(ctx, team, issue.Poster, true, issue)
+				if err != nil {
+					return err
+				}
+				_, err = issue_service.TeamReviewRequest(ctx, issue, issue.Poster, team, true)
+				if err != nil {
+					return err
+				}
+				continue
+			}
+
+			reviewer, err := user_model.GetUserByID(ctx, reviewerID)
+			if err != nil {
+				return err
+			}
+			permDoer, err := access_model.GetUserRepoPermission(ctx, issue.Repo, issue.Poster)
+			if err != nil {
+				return err
+			}
+			err = issue_service.IsValidReviewRequest(ctx, reviewer, issue.Poster, true, issue, &permDoer)
+			if err != nil {
+				return err
+			}
+			_, err = issue_service.ReviewRequest(ctx, issue, issue.Poster, reviewer, true)
```

Comment:
This also creates the review request notification. So the user will receive the "review requested" notification before the "pull request created" notification. If you look at the assignee code, they are first added without sending a notification. The notification is sent later on: https://github.com/go-gitea/gitea/blob/62d8403785c916c47df050eec6a2be20a931f0c1/services/pull/pull.go#L198

Due to this, it also seems like the UI notification cannot be created. I'm seeing this in the logs:

`2023/12/30 13:20:58 ...tification/notify.go:56:handler() [E] Was unable to create issue notification: issue does not exist [id: 10, repo_id: 0, index: 0]`

I suspect this is because the notifier runs in a different transaction and does not "see" the newly created PR yet?